meson.build

```diff
@@ -1608,6 +1608,21 @@ have = get_option('sysupdate').require(
         conf.get('HAVE_LIBFDISK') == 1,
         error_message : 'fdisk and openssl required').allowed()
 conf.set10('ENABLE_SYSUPDATE', have)
+if have
+        sysupdate_mode = get_option('sysupdate-mode')
+        if sysupdate_mode == 'default'
+                sysupdate_mode = 'sysupdated'
+                # TODO: Should the default be both? Packagers will need to split
+                #       out sub-packages by default then. Maybe default should
+                #       cause a build failure instead, so you _have_ to pick one?
+        endif
+
+        conf.set10('ENABLE_SYSUPDATED', sysupdate_mode == 'sysupdated' or sysupdate_mode == 'both')
+        conf.set10('ENABLE_SYSUPDATE_UNITS', sysupdate_mode == 'units' or sysupdate_mode == 'both')
```

Comment:
This seems iffy. In the commit message you say "systemd-sysupdated (the dbus service) conflicts with the existing basic .service/.timer units". But what does this mean? At install time (I guess not, since "both" is an option). So this means that they conflict at runtime. That means that there's no reason to not install both. This option can be dropped.

Reply:
Yeah at runtime. The way things currently are, distros shouldn't ship both because they conflict with each other at runtime.

We can fix this eventually by just making the units talk to sysupdated. But then it won't make much sense to have the switch available at all, since now you'd need sysupdated to have the basic sysupdate functionality work

Reply:
This is a misunderstanding. If they conflict at runtime, there is not reason for distros to not ship them together.

Reply:
Yeah the intent was for distros to set it to `both` and then make subpackages, then more specialized usecases (like GNOME OS or embedded IOT images) would decide which they want to use

Reply:
Actually there is no need to create subpackages.

Reply:
Is there some way to make enabling one unit disable another that I don't know about? Because enabling the sysupdate timer should disable sysupdated (and vice versa). This can be done in packaging via subpackages that conflict with each other.

Reply:
Packages/subpackages manage units by calling scriptlets. So by definition, everything that packages do can also be done called separately.  We can declare Conflicts on units, but we can do that without creating subpackages.

Having subpackages conflict is not really a solution. It's very annoying to users and is not a preferred mechanism for switching between implementations.

Reply:
I didn't mean using scriptlets. I mean having a subpackage that ships the systemd-sysupdated binaries + units + dbus config + polkit config + etc, and a separate subpackage that ships the sysupdate basic unit files.

But again, this is all beside the point because we'll eventually just make the units use sysupdated. Then we can probably remove the meson option controlling it also (since, at that point, it won't make sense to turn it off independently from sysupdate)

Reply:
Oh man. Just having separate packages doesn't do anything magically by itself. How could it? I'm very much confused as to how you think this all works.

But yeah, I agree that we don't need to discuss this now.